### ISSUE TITLE
rpminspect reports bogus errors with the xml test on html files in the appstream package.

### ISSUE THREAD

**Issue by mike-fabian** [rpminspect/rpminspect#449]
I download the latest appstream packages for fedora using:

```
rpminspect-fedora -v --fetch-only appstream-0.14.3-2.fc34
```

Then I run the test for xml:

```
rpminspect-fedora --tests=xml appstream-0.14.3-2.fc34
```

This reports problems in .html files in the documentation of appstream. For  example:

```
xml-files:
----------
1) File /usr/share/doc/appstream/html/chap-AppStream-API.html is a malformed XML file on
x86_64 
Result: VERIFY
Waiver Authorization: Anyone

Details:
No declaration for attribute xmlns:dm of element span


Suggested Remedy: Correct the reported errors in the XML document
```

The file does contain stuff like:

```
</span><span xmlns:dm="urn:x-suse:ns:docmanager" class="name">AppStream API Reference</span>
```

But that is probably OK.

xmllint does not complain when checking the same file.

**Comment by hadess** [comment]
I have the same problem with `rpminspect version: 1.6-0.1.202106081929git.fc34 (with data package: 1:1.5-0.1.202105061811git.fc34)` yelling that 50 (valid) docbook XML files are invalid in the `grilo-plugins` package.

```
xml:
----

1) File /usr/share/help/C/grilo-plugins/grilo-plugins.xml is a malformed XML file on aarch64

Result: VERIFY
Waiver Authorization: Anyone

Details:
No declaration for element book


Suggested Remedy:
Correct the reported errors in the XML document
[...]
50) File /usr/share/help/uk/grilo-plugins/grilo-plugins.xml is a malformed XML file on s390x

Result: VERIFY
Waiver Authorization: Anyone

Details:
No declaration for element book


Suggested Remedy:
Correct the reported errors in the XML document


Error: exec session exited with non-zero exit code 1: OCI runtime error
```

Unfortunately the failing message doesn't include the command-line used to run this test, or the actual error message generated when running this test, so the test can't be reproduced or debugged outside of rpminspect.

**Comment by dcantrell** [comment]
This is now fixed with the commit I just made for issue #455 

rpminspect now reports VERIFY results for XML that is not well-formed, but invalid XML is just reported as an INFO result which does not cause a failure exit code.

librpminspect uses libxml2 directly, it does not invoke xmllint to check XML files.

**Comment by hadess** [comment]
>         doc = xmlCtxtReadFile(ctxt, path, NULL, XML_PARSE_RECOVER | XML_PARSE_NONET);

If rpminspect doesn't have access to the network (it probably shouldn't) and you load XML files with `NONET`, that means that DTDs referenced from the network are never loaded. So that code is checking for XML files being valid against a DTD it never loads. It just can't work.

**Comment by dcantrell** [comment]
> > ```
> >     doc = xmlCtxtReadFile(ctxt, path, NULL, XML_PARSE_RECOVER | XML_PARSE_NONET);
> > ```
> 
> If rpminspect doesn't have access to the network (it probably shouldn't) and you load XML files with `NONET`, that means that DTDs referenced from the network are never loaded. So that code is checking for XML files being valid against a DTD it never loads. It just can't work.

Network access is not assumed, so I have tried to make sure everything can run with resources on the system.  I know NONET will break for anything that needs to fetch a DTD online, but I couldn't find any only useful guidance.  Fortunately this bug showed up and I was able to dig in to it a bit more and now librpminspect makes the distinction between well-formed XML and valid XML for reporting purposes.  So if xmlCtxtReadFile() gives an error, that will result in a VERIFY result.  If it fails to parse and generates an error, then rpminspect reports that as INFO but it doesn't cause a failure.  Given that I have to assume network access will not be available and the fact that a lot of XML things are shipped and are deliberately invalid (e.g., a test suite) then the only real thing we can do here is ensure it's all well-formed.  So that's what I've done now.

Thanks.
